From 2c96fcfe5037e1078c24f824a0c84050cf54532b Mon Sep 17 00:00:00 2001
From: Mike Wester <mwester@dis.net>
Date: Sun, 13 Apr 2008 07:25:56 +0100
Subject: [PATCH] fix-usb-ethernet-mwester.patch

This is a kernel bug affecting all platforms, introduced in some code cleanup in
2.6.23.  Patch will be posted as soon as I test it.

NOTE: Do not submit upstream; this patch or something very similar is already
in process for 2.6.25.
---
 drivers/usb/gadget/ether.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/gadget/ether.c b/drivers/usb/gadget/ether.c
index 6efbb44..758fd27 100644
--- a/drivers/usb/gadget/ether.c
+++ b/drivers/usb/gadget/ether.c
@@ -1558,6 +1558,7 @@ done_set_intf:
 				memcpy(req->buf, buf, n);
 				req->complete = rndis_response_complete;
 				rndis_free_response(dev->rndis_config, buf);
+				value = n;
 			}
 			/* else stalls ... spec says to avoid that */
 		}
-- 
1.5.6.5

